### PR TITLE
Zero out the AWS credential for CC upgrade

### PR DESCRIPTION
**Pull request by 12345lcr**
### What this PR does / why we need it
This is to ensures that the Cluster API Provider AWS controller is pinned to control plane nodes and is running without static credentials such that Cluster API AWS runs using the EC2 instance profile attached to the control plane node. 

This is done by zeroing out the credentials secret for CAPA, causing the AWS SDK to fall back to the default credential provider chain. 

This was implemented for mgmt cluster creation and legacy mgmt cluster upgrade, but missing in classy mgmt cluster upgrade.

Additionally extend the k8s versionpatch waiting time to allow corner cases.

### Which issue(s) this PR fixes

Fixes #

### Release note

```release-note

```

### CODE REVIEW

**Comment by alfredthenarwhal** [comment]
> @12345lcr:  /test upgrade-aws
Commit: 33e9466

Tests failed! Build no: 4593

File: tkg/client/upgrade_cluster_clusterclass.go [hunk 1]
```diff
@@ -16,15 +16,20 @@ import (
 func (c *TkgClient) DoClassyClusterUpgrade(regionalClusterClient clusterclient.Client,
 	currentClusterClient clusterclient.Client, options *UpgradeClusterOptions) error {
 
+	if err := currentClusterClient.PatchClusterAPIAWSControllersToUseEC2Credentials(); err != nil {
```
Comment (DanielXiao) [inline tkg/client/upgrade_cluster_clusterclass.go:19]: Should it be executed right after provider upgrade?

Comment (DanielXiao) [reply tkg/client/upgrade_cluster_clusterclass.go:19]: Right after https://github.com/vmware-tanzu/tanzu-framework/blob/fe8dd986a4313e34518dcea3d7025c3b8c2e84e0/tkg/client/upgrade_region.go#L135

**Comment by 12345lcr** [comment]
Close since https://github.com/vmware-tanzu/tanzu-framework/pull/4531 is merged
